src/main/scala/com/databricks/labs/mosaic/expressions/raster/RST_FromContent.scala

```diff
@@ -0,0 +1,136 @@
+import org.apache.spark.sql.catalyst.expressions.{CollectionGenerator, Expression, Literal, NullIntolerant}
+import org.apache.spark.sql.types.{DataType, IntegerType, StringType, StructField, StructType}
+import org.apache.spark.unsafe.types.UTF8String
+
+import java.nio.file.{Files, Paths, StandardOpenOption}
+
+/**
+  * The raster for construction of a raster tile. This should be the first
+  * expression in the expression tree for a raster tile.
+  */
+case class RST_FromContent(
+    rasterExpr: Expression,
+    driverExpr: Expression,
+    parentPathExpr: Expression,
+    sizeInMB: Expression,
+    expressionConfig: MosaicExpressionConfig
+) extends CollectionGenerator
+      with Serializable
+      with NullIntolerant
+      with CodegenFallback {
+
+    override def dataType: DataType = RasterTileType(expressionConfig.getCellIdType)
+
+    protected val geometryAPI: GeometryAPI = GeometryAPI.apply(expressionConfig.getGeometryAPI)
+
+    protected val indexSystem: IndexSystem = IndexSystemFactory.getIndexSystem(expressionConfig.getIndexSystem)
+
+    protected val cellIdDataType: DataType = indexSystem.getCellIdDataType
+
+    override def position: Boolean = false
+
+    override def inline: Boolean = false
+
+    override def children: Seq[Expression] = Seq(rasterExpr, driverExpr, parentPathExpr, sizeInMB)
+
+    override def elementSchema: StructType = StructType(Array(StructField("tile", dataType)))
+
+    /**
+      * subdivides raster binary content into tiles of the specified size (in MB).
+      * @param input
+      *   The input file path.
+      * @return
+      *   The tiles.
+      */
+    override def eval(input: InternalRow): TraversableOnce[InternalRow] = {
+        GDAL.enable(expressionConfig)
+        //parentPath may be null (it is not used here as content may be different)
+        val parentPath = {
+            try {
+                parentPathExpr.eval(input).asInstanceOf[UTF8String].toString
+            } catch {
+                case _: Any => null
+            }
+        }
+        val driver = driverExpr.eval(input).asInstanceOf[UTF8String].toString
+        val ext = GDAL.getExtension(driver)
+        var rasterArr = rasterExpr.eval(input).asInstanceOf[Array[Byte]]
+        val targetSize = sizeInMB.eval(input).asInstanceOf[Int]
+        if (targetSize <= 0 && rasterArr.length <= Integer.MAX_VALUE) {
+            var raster = MosaicRasterGDAL.readRaster(rasterArr, parentPath, driver)
+            var tile = MosaicRasterTile(null, raster, parentPath, driver)
+            val row = tile.formatCellId(indexSystem).serialize()
+            RasterCleaner.dispose(raster)
+            RasterCleaner.dispose(tile)
+            rasterArr = null
+            raster = null
+            tile = null
+            Seq(InternalRow.fromSeq(Seq(row)))
+        } else {
+            // If target size is <0 and we are here that means the file is too big to fit in memory
```

Comment:
@mjohns-databricks I dont think this branch can ever happen in this context.
I can occur only if we are directly reading the files.
Binary format will die on files larger than 2.3 GB.
The way around would be to mark files are splitable and have a SeqNo and then merge them in a group by, read them then split them into tiles (very messy :/)